platform/jvm/replay/src/main/kotlin/io/bitdrift/capture/replay/internal/ScreenshotCaptureEngine.kt

```diff
@@ -0,0 +1,165 @@
+                finishOnError(
+                    expected = false,
+                    "Screenshot triggered: PixelCopy compression failed. Exception=${e.message}",
+                    e
+                )
+            } finally {
+                resultBitmap.recycle()
+            }
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun pixelCopySnapshot(root: View) {
+        val window = root.phoneWindow
+        if (window == null) {
+            finishOnError(expected = true, "Screenshot triggered: Phone window invalid, skipping capture")
+            return
+        }
+
+        val resultBitmap = Bitmap.createBitmap(
+            root.width,
+            root.height,
+            Bitmap.Config.RGB_565
+        )
+        // TODO(murki): Figure out if there's any benefit from calling this using mainThreadHandler.mainHandler.post{ }
+        PixelCopy.request(
+            window,
+            resultBitmap,
+            { screenshotResultStatus: Int ->
+                if (screenshotResultStatus != PixelCopy.SUCCESS) {
+                    resultBitmap.recycle()
+                    finishOnError(
+                        expected = false,
+                        "Screenshot triggered: PixelCopy operation failed. Result.status=${screenshotResultStatus.toStatusText()}",
+                    )
+                    return@request
+                }
+
+                // TODO(murki): Try to avoid so much context switching between main and background threads
+                executor.execute {
+                    try {
+                        metrics.screenshot(resultBitmap.allocationByteCount, resultBitmap.byteCount)
+                        val screenshotBytes = compressScreenshot(resultBitmap)
+                        metrics.compression(screenshotBytes.size)
+                        logger.onScreenshotCaptured(screenshotBytes, metrics.data())
+                    } catch (e: Exception) {
+                        finishOnError(
+                            expected = false,
+                            "Screenshot triggered: Compression operation failed. Exception=${e.message}",
+                            e
+                        )
+                    } finally {
+                        resultBitmap.recycle()
+                    }
+                }
+
+            },
+            mainThreadHandler.mainHandler,
+        )
+    }
+
+    private fun finishOnError(expected: Boolean, message: String, e: Throwable? = null) {
+        if (!expected) {
+            errorHandler.handleError(message, e)
+        }
+        logger.logErrorInternal(message, e)
+        // Log empty screenshot on unblock the rust engine caller
+        logger.onScreenshotCaptured(ByteArray(0), metrics.data())
+    }
+
+    private fun compressScreenshot(resultBitmap: Bitmap): ByteArray {
+        return ByteArrayOutputStream().use { outStream ->
+            resultBitmap.compress(Bitmap.CompressFormat.JPEG, 10, outStream)
```

Comment:
have we tried making this `0`?

Reply:
If this is using full size with device's pixel density I'd also consider resizing it

Reply:
you can see the result images in the linked sessions, the metadata also contains the used bytes and time spent on each operation